src/common/valnum.cpp

```diff
@@ -234,9 +234,29 @@ wxString wxIntegerValidatorBase::ToString(LongestValueType value) const
 }
 
 bool
-wxIntegerValidatorBase::FromString(const wxString& s, LongestValueType *value)
+wxIntegerValidatorBase::FromString(const wxString& s,
+                                   LongestValueType *value) const
 {
-    return wxNumberFormatter::FromString(s, value);
+    if ( CanBeNegative() )
+    {
+        return wxNumberFormatter::FromString(s, value);
+    }
+    else
+    {
+        // Parse as unsigned to ensure we don't accept minus sign here.
+#ifdef wxULongLong_t
+        wxULongLong_t uvalue;
+#else
+        wxULongLong_t uvalue;
+#endif
```

Comment:
Looks like we should do this instead:
```
#ifdef wxHAS_LONG_LONG_T_DIFFERENT_FROM_LONG
        wxULongLong_t uvalue;
#else
        unsigned long uvalue;
#endif
```
to be consistent with what is already done by the library itself elsewhere!

Reply:
Oops, yes, this is what this was supposed to be, thanks for noticing.